Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/shape_smoothing.h

```diff
@@ -0,0 +1,415 @@
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      std::cout << "setup_system... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.setup_system(A, bx, by, bz, stiffness, time);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      std::cout << "solve_system... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.solve_system(A, Xx, Xy, Xz, bx, by, bz);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      std::cout << "update_mesh... ";
+      t.reset(); t.start();
+      #endif
+
+      smoother.update_mesh(Xx, Xy, Xz);
+
+      #ifdef CGAL_PMP_SMOOTHING_VERBOSE
+      t.stop();
+      std::cout << " done ("<< t.time() <<" sec)." << std::endl;
+      #endif
+    }
+  }
+}
+
+template<typename PolygonMesh, typename NamedParameters>
+void smooth_along_curvature_flow(PolygonMesh& pmesh, const double& time,
+                                 const NamedParameters& np)
+{
+  smooth_along_curvature_flow(faces(pmesh), pmesh, time, np);
+}
+
+template<typename PolygonMesh>
+void smooth_along_curvature_flow(PolygonMesh& pmesh, const double& time)
+{
+  smooth_along_curvature_flow(faces(pmesh), pmesh, time,
+                              parameters::all_default());
+}
+
+// API for Polyhedron demo plugin
+namespace internal{
+
+template<typename PolygonMesh, typename FaceRange, typename NamedParameters>
+void solve_mcf(const FaceRange& faces, PolygonMesh& mesh, const double& time,
+               std::vector<CGAL::Triple<int, int, double> >& stiffness, bool compute_stiffness,
+               const NamedParameters& np)
+{
+  using boost::choose_param;
+  using boost::get_param;
+
+  //typedef typename GetGeomTraits<PolygonMesh>::type GeomTraits;
+  //typedef typename boost::property_map<PolygonMesh, CGAL::vertex_point_t>::type VertexPointMap;
+  //VertexPointMap vpmap = get(CGAL::vertex_point, mesh);
+
+  typedef typename GetGeomTraits<PolygonMesh, NamedParameters>::type GeomTraits;
+
+  typedef typename GetVertexPointMap<PolygonMesh, NamedParameters>::type VertexPointMap;
+  VertexPointMap vpmap = choose_param(get_param(np, internal_np::vertex_point),
+                               get_property_map(CGAL::vertex_point, mesh));
+
+  typedef typename boost::graph_traits<PolygonMesh>::vertex_descriptor vertex_descriptor;
+  typedef typename boost::lookup_named_param_def <
+      internal_np::vertex_is_constrained_t,
+      NamedParameters,
+      internal::Constrained_vertices_map<vertex_descriptor>
+    > ::type VCMap;
+  VCMap vcmap = choose_param(get_param(np, internal_np::vertex_is_constrained),
+                             internal::Constrained_vertices_map<vertex_descriptor>());
+
+  // nb_iterations
+  std::size_t nb_iterations = choose_param(get_param(np, internal_np::number_of_iterations), 1);
+
+#if defined(CGAL_EIGEN3_ENABLED)
+  #if EIGEN_VERSION_AT_LEAST(3,2,0)
```

Comment:
same issue here. This is an internal function and the solver must be passed as parameter to the function.